corehq/apps/geospatial/static/geospatial/js/geospatial_map.js

```diff
@@ -169,29 +170,98 @@ hqDefine("geospatial/js/geospatial_map", [
             return self;
         };
 
-        var exportGeoJson = function(drawInstance) {
+        var exportGeoJson = function(drawInstance, mapControlsModelInstance) {
             // Credit to https://gist.github.com/danswick/36796153bd86ce982a59043cbe0ac8f7
             // I could not get this to work using knockout.js. It did set the attributes, but a download wasn't
             // triggered
             var exportButton = $("#btnExport");
             var data = drawInstance.getAll();
 
             if (data.features.length) {
-                // Stringify the GeoJson
-                var convertedData = 'text/json;charset=utf-8,' + encodeURIComponent(JSON.stringify(data));
-
-                // Create export
-                exportButton.attr('href', 'data:' + convertedData);
-                exportButton.attr('download','data.geojson');
+                let name = window.prompt("Name of the Area");
+                data['name'] = name;
+
+                $.ajax({
+                    type: 'post',
+                    url: saveGeoJSONUrl,
+                    dataType: 'json',
+                    data: JSON.stringify({'geo_json': data}),
+                    contentType: "application/json; charset=utf-8",
+                    success: function (ret) {
+                        delete data.name;
+                        // delete drawn area
+                        drawInstance.deleteAll();
```

Comment:
Why is it necessary to redraw the polygon?

Reply:
The draws and Polygons are different in terms of the data models. The drawn objects are managed by the Mapbox Draw Instance, whereas the saved polygon objects are managed by our code. Without redrawing, the user will have option to keep saving the same polygon